rsmtool/rsmtool.py

```diff
@@ -65,24 +72,34 @@ def run_experiment(config_file_or_obj,
     makedirs(figdir, exist_ok=True)
     makedirs(reportdir, exist_ok=True)
 
-    # Allow users to pass Configuration object to the
-    # `config_file_or_obj` argument, rather than read from file
-    if not isinstance(config_file_or_obj, Configuration):
+    # check what sort of input we got
+    if isinstance(config_file_or_obj_or_dict, str):
 
         # Instantiate configuration parser object
-        parser = ConfigurationParser.get_configparser(config_file_or_obj)
-        configuration = parser.read_normalize_validate_and_process_config(config_file_or_obj)
+        parser = ConfigurationParser.get_configparser(config_file_or_obj_or_dict)
+        configuration = parser.read_normalize_validate_and_process_config(config_file_or_obj_or_dict)
 
         # get the directory where the configuration file lives
-        configpath = dirname(config_file_or_obj)
+        configpath = dirname(config_file_or_obj_or_dict)
+
+    elif isinstance(config_file_or_obj_or_dict, dict):
+
+        # initialize the parser from dict
+        parser = ConfigurationParser()
+        parser.load_config_from_dict(config_file_or_obj_or_dict)
+        configuration = parser.normalize_validate_and_process_config()
+        # set configpath to current directory
+        configpath = getcwd()
+        configuration.filepath = configpath
 
     else:
```

Comment:
Is there a reason not to _also_ check `isinstance(config_file_or_obj_or_dict, Configuration)`? In some cases where I've had this sort of `if`/`elif`/`else`, I like to make sure about what I'm interacting with (or try to at least) in each piece and then leave the `else` part to deal with an "unrecognized" value-type error and raise an exception.

Reply:
Good point. Also added a test for this.